### PR TITLE
feat: add docker support.

### PR DESCRIPTION
**Pull request by ayamir**
I built a docker image for `nvimdots`: https://hub.docker.com/repository/docker/ayamir/nvimdots/general

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+# Install the ca-certificates package to update CA certificates
+RUN apt-get update && apt-get install -y \
+    ca-certificates
+
+# Update CA certificates
+RUN update-ca-certificates
+
+# Add a command to select a mirror based on a condition
+ARG use_mirror=false
```
Comment (aarnphm) [inline Dockerfile:16]: more docker-ich

```suggestion
ARG USE_MIRROR=false
ENV use_mirror=$USE_MIRROR
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+RUN git clone https://github.com/neovim/unibilium.git && \
+    cd unibilium && \
+    autoreconf -fi && \
+    ./configure --prefix=/usr/local && \
+    make && \
+    make install && \
+    cd ../ && rm -rf unibilium
+
+# Install the latest stable neovim
+RUN curl -LO https://github.com/neovim/neovim/releases/download/stable/nvim-linux64.tar.gz \
```
Comment (aarnphm) [inline Dockerfile:84]: should we include two variants? One stable and one nightly?

Probably can use multi-build here.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+WORKDIR /root
+
+# Set TERM
+ENV TERM=xterm-256color
+
+# Bootstrap
+RUN nvim --headless "+Lazy! sync" +qa
+
+# Start with bash for the convenience of new changes
+CMD ["bash"]
```
Comment (aarnphm) [inline Dockerfile:125]: For publishing to ghcr I recommend to also setup a actions to build for both arm and amd64

**Review by aarnphm — COMMENTED** [review]
few tings

```bash

docker run --rm -it -v ${PWD}:/data ghcr.io/ayamir/nvimdots:nightly

docker run --rm -it -v ${PWD}:/data ghcr.io/anymir/nvimdots:stable
```

**Review by CharlesChiuGit — APPROVED** [review]
Other than what aarnphm had already pointed out, i think it's go to go as a first draft.

**Comment by ayamir** [comment]
close because it seems meaningless...

**Pull request closed without merging**